src/components/member-page/donation/table/row.js

```diff
@@ -0,0 +1,208 @@
+  } = record
+  const donationDate = formattedDate(new Date(createdAt))
+  const amountSuffix = type === DonationType.PRIME ? '元' : '元/月'
+  const [isOpen, setIsOpen] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [periodicHistory, setPeriodicHistory] = useState([])
+  const { releaseBranch } = useContext(CoreContext)
+  const [state, dispatch] = useStore()
+
+  const getPeriodicDonationHistory = () => {
+    setIsLoading(true)
+    const detail = _.get(state, [
+      reduxStateFields.donationHistory,
+      'periodicDonationHistory',
+      'records',
+      orderNumber,
+    ])
+    if (detail) {
+      setPeriodicHistory(detail.records)
+    } else {
+      const jwt = _.get(state, [reduxStateFields.auth, 'accessToken'])
+      dispatch(getUserPeriodicDonationHistory(jwt, orderNumber, 0, 18)).then(
+        res => {
+          setPeriodicHistory(_.get(res, 'payload.data.records'))
+        }
+      )
+    }
+    setTimeout(() => setIsLoading(false), 1000)
+  }
+
+  const handleRowClick = () => {
+    if (type !== DonationType.PERIODIC) return
+    if (!isOpen) {
+      getPeriodicDonationHistory()
+    }
+    setIsOpen(!isOpen)
+  }
+  return (
+    <React.Fragment>
+      <TableContent onClick={handleRowClick}>
+        <P1Gray800 className="donation-date" text={donationDate} />
+        <P1Gray800 className="type" text={DonationTypeLabel[type]} />
+        <P1Gray800 className="donation-number" text={orderNumber} />
+        <P1Gray800
+          className="amount"
+          text={`${amount.toLocaleString('en-US')}${amountSuffix}`}
+        />
+        <div className="status">
+          <StatusBadge status={status} type={type} />
+          {type === DonationType.PERIODIC && (
+            <ArrowIcon
+              direction="down"
+              rotate={isOpen ? '-180deg' : '0'}
+              releaseBranch={releaseBranch}
+            />
+          )}
+        </div>
+      </TableContent>
+      <MobileTableContent onClick={handleRowClick}>
+        <P1Gray800 text={donationDate} />
+        <div className="row">
+          <P1Gray800 weight={P1.Weight.BOLD} text={DonationTypeLabel[type]} />
+          <StatusBadge status={status} type={type} />
+          <P1Gray800
+            className="amount"
+            text={`${amount.toLocaleString('en-US')}${amountSuffix}`}
+          />
+          {type === DonationType.PERIODIC && (
+            <ArrowIcon
+              direction="down"
+              rotate={isOpen ? '-180deg' : '0'}
+              releaseBranch={releaseBranch}
+            />
+          )}
+        </div>
+        <P1Gray600 className="donation-number" text={orderNumber} />
+      </MobileTableContent>
+      {isOpen && (
+        <TableRowDetail
+          record={record}
+          periodicHistory={periodicHistory}
+          isLoading={isLoading}
+        />
+      )}
+      <Divider />
+    </React.Fragment>
+  )
+})
+
+TableRow.propTypes = {
+  record: PropTypes.object.isRequired,
```

Comment:
same as [this comment](https://github.com/twreporter/twreporter-react/pull/2823/files#r1568308036)

Reply:
It will be implemented after the spec is confirmed.